content/oci/oci.go

```diff
@@ -398,6 +402,42 @@ func (s *Store) writeIndexFile() error {
 	return os.WriteFile(s.indexPath, indexJSON, 0666)
 }
 
+// GC removes garbage from Store. The garbage to be cleaned are:
+//   - garbage blobs in the storage whose metadata is not stored in Store.
+//   - unreferenced (dangling) blobs in Store which have no predecessors
+func (s *Store) GC(ctx context.Context) error {
+	s.sync.RLock()
+	defer s.sync.RUnlock()
+
+	rootpath := filepath.Join(s.root, "blobs", "sha256")
+	return filepath.Walk(rootpath, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		// skip the working directory
+		if !info.IsDir() {
+			blobDigest := digest.NewDigestFromEncoded("sha256", info.Name())
+			exists, desc := s.graph.ExistsInStore(blobDigest)
+			if !exists {
+				// remove the blob from storage if it does not exist in Store
+				err = os.Remove(path)
+				if err != nil {
+					if errors.Is(err, fs.ErrNotExist) {
+						return fmt.Errorf("%s: %w", blobDigest, errdef.ErrNotFound)
+					}
+					return err
+				}
+			} else if !s.graph.HasPredecessors(blobDigest) {
```

Comment:
Looks like `HasPredecessors` is not necessary?
We can get a descriptor from the memory graph via the digest, and then we can call `Predecessors` and check the size of the result array.

Reply:
Removed this function after refining the GC logic.